modules/utxo-lib/src/bitgo/zcash/ZcashPsbt.ts

```diff
@@ -128,9 +129,17 @@ export class ZcashPsbt extends UtxoPsbt<ZcashTransaction<bigint>> {
   // transactions because zcash hashes the value directly. Thus, it is unnecessary to have
   // the previous transaction hash on the unspent.
   signInput(inputIndex: number, keyPair: Signer, sighashTypes?: number[]): this {
+    return this.withUnsafeSignNonSegwitTrue(super.signInput.bind(this, inputIndex, keyPair, sighashTypes));
+  }
+
+  validateSignaturesOfInput(inputIndex: number, validator: ValidateSigFunction, pubkey?: Buffer): boolean {
+    return this.withUnsafeSignNonSegwitTrue(super.validateSignaturesOfInput.bind(this, inputIndex, validator, pubkey));
+  }
+
+  private withUnsafeSignNonSegwitTrue(callback: () => any): any {
     (this as any).__CACHE.__UNSAFE_SIGN_NONSEGWIT = true;
     try {
-      return super.signInput(inputIndex, keyPair, sighashTypes);
+      return callback();
```

Comment:
hate to be that guy, but callback is a confusing name here. `fn`, `func`, `call`, something that doesn't already have an accepted meaning different from what we're doing.